### PR TITLE
feat(rollouts): Add CRUD rollouts and new evaluation operations

### PR DESCRIPTION
**Pull request by yquansah**
This PR does the following:
- Adds CRUD operations for new rollout endpoints for evaluation v2 feature
- Adds new evaluation operations for (variant, boolean, batch)

Completes FLI-425

### CODE REVIEW

**Comment by yquansah** [comment]
ITs will fail until evaluation v2 feature reaches the `flipt:latest` image.

File: src/api/evaluation.rs [hunk 1]
```diff
@@ -33,6 +33,24 @@ impl<'client> EvaluationClient<'client> {
         );
         self.client.post(&path, Some(batch)).await
     }
+
+    pub async fn boolean(&self, eval: &V2EvaluateRequest) -> Result<BooleanEvaluation> {
```
Comment (markphelps) [inline src/api/evaluation.rs:37]: can we put these in their own module? that way we don't need to name the types `V2XYZ`. Its a bit confusing calling them `v2` IMO when in the API route its `/evaluate/v1`.

I think if we create a folder at `src/evaluate` then we can move all this new code there without the names coliding

File: src/api/rollout.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                .as_ref()
+                .unwrap_or(&DEFAULT_NAMESPACE.to_string()),
+            flag_key = delete.flag_key,
+            id = delete.id
+        );
+
+        self.client.delete(&path, None::<&()>).await
+    }
+
+    pub async fn update(&self, update: &RolloutUpdateRequest) -> Result<Rollout> {
```
Comment (markphelps) [inline src/api/rollout.rs:55]: missing the `order` API impl

`pub async fn order`

**Review by markphelps — CHANGES_REQUESTED** [review]
lookin good

File: src/api/evaluation.rs (hunk 1 above)
Comment (yquansah) [reply src/api/evaluation.rs]: @markphelps Yeah, that makes sense so we can avoid all that prefixing. Thanks!

File: src/api/rollout.rs (hunk 2 above)
Comment (yquansah) [reply src/api/rollout.rs:55]: @markphelps Thanks, will add that!

File: src/api/evaluation.rs (hunk 1 above)
Comment (markphelps) [reply src/api/evaluation.rs]: I still think moving the new evaluation API to its own folder like we have for `auth` and `meta` would make the most sense, basically move the new `evaluate.rs` to `src/evaluate/mod.rs`

Comment (markphelps) [reply src/api/evaluation.rs]: @yquansah wdyt

Comment (yquansah) [reply src/api/evaluation.rs]: @markphelps Ah okay that is what you meant. Yeah I agree, since it is its own endpoint `/evaluate/v1`. That makes sense.

**Comment by markphelps** [comment]
we should prob wait until `latest` gets updated today with the release of v1.24, then can ensure the ITs pass in CI

File: src/api/flag.rs [hunk 3]
```diff
@@ -89,6 +89,7 @@ pub struct FlagCreateRequest {
     pub name: String,
     pub description: String,
     pub enabled: bool,
+    pub r#type: FlagType,
```
Comment (markphelps) [inline src/api/flag.rs:92]: i think we should make this optional so that it is backward compatible

**Review by markphelps — CHANGES_REQUESTED** [review]

File: src/api/flag.rs (hunk 3 above)
Comment (yquansah) [reply src/api/flag.rs:92]: @markphelps great catch!

File: src/evaluate/mod.rs [hunk 4]
```diff
@@ -0,0 +1,125 @@
+use crate::api::{ApiClient, Result};
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+pub struct EvaluateClient<'client> {
```
Comment (markphelps) [inline src/evaluate/mod.rs:6]:
```suggestion
pub struct EvaluationClient<'client> {
```

wdyt about `%s/Evaluate/Evaluation/g` here? I think the module name can stay the same, but just for consistency maybe use `Evaluation` for all types?

**Review by markphelps — COMMENTED** [review]
minor naming request, looking great though!

**Review by brettbuddin — APPROVED** [review]

**Pull request merged**